### PR TITLE
Split test/test_components.py into test/components/test_*.py

### PR DESCRIPTION
**Pull request by whitphx**
## Description

Refactoring, based on [this context](https://github.com/gradio-app/gradio/pull/8041#discussion_r1578419523).

Closes: #8125

### CODE REVIEW

**Review by abidlabs — APPROVED** [review]
Lovely. Thanks for taking care of this @whitphx!

File: test/components/conftest.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import os
```
Comment (freddyaboulton) [inline test/components/conftest.py:1]: This is already set in the `pyproject.toml` file

**Review by freddyaboulton — APPROVED** [review]
Thanks @whitphx !

**Comment by whitphx** [comment]
Added tests were run successfully.

https://github.com/gradio-app/gradio/actions/runs/8980681841/job/24664770460

```
test/components/test_annotated_image.py ....                             [ 82%]
test/components/test_audio.py ......                                     [ 83%]
test/components/test_bar_plot.py ......                                  [ 83%]
test/components/test_button.py .                                         [ 84%]
test/components/test_chatbot.py ..                                       [ 84%]
test/components/test_checkbox.py ..                                      [ 84%]
test/components/test_checkbox_group.py ..                                [ 84%]
test/components/test_code.py ..                                          [ 85%]
test/components/test_color_picker.py ....                                [ 85%]
test/components/test_dataframe.py .....                                  [ 86%]
test/components/test_dataset.py ...                                      [ 86%]
test/components/test_dropdown.py ..                                      [ 87%]
test/components/test_file.py .....                                       [ 87%]
test/components/test_file_explorer.py ..                                 [ 88%]
test/components/test_gallery.py ....                                     [ 88%]
test/components/test_highlighted_text.py ...                             [ 89%]
test/components/test_html.py ..                                          [ 89%]
test/components/test_image.py .....                                      [ 90%]
test/components/test_image_editor.py ..                                  [ 90%]
test/components/test_json.py .........                                   [ 91%]
test/components/test_label.py ...                                        [ 91%]
test/components/test_line_plot.py ......                                 [ 92%]
test/components/test_markdown.py ..                                      [ 93%]
test/components/test_model3d.py ..                                       [ 93%]
test/components/test_number.py .........                                 [ 94%]
test/components/test_plot.py ....                                        [ 95%]
test/components/test_radio.py ..                                         [ 95%]
test/components/test_scatter_plot.py .........                           [ 96%]
test/components/test_slider.py .....                                     [ 97%]
test/components/test_state.py ....                                       [ 97%]
test/components/test_textbox.py .......                                  [ 98%]
test/components/test_upload_button.py ...                                [ 99%]
test/components/test_video.py .....                                      [100%]

```

**Pull request merged**